go-controller/pkg/ovn/endpoints.go

```diff
@@ -281,26 +276,27 @@ func (ovn *Controller) deleteEndpoints(ep *kapi.Endpoints) error {
 		var lb string
 		lb, err = ovn.getLoadBalancer(svcPort.Protocol)
 		if err != nil {
-			klog.Errorf("Failed to get load-balancer for %s (%v)",
-				lb, err)
+			klog.Errorf("Failed to get load-balancer for %s (%v)", lb, err)
 			continue
 		}
 
-		quotedHostPort := "\"" + util.JoinHostPortInt32(svc.Spec.ClusterIP, svcPort.Port) + "\""
-		if keepEmptyLB(svc) {
-			key := "vips:" + quotedHostPort + "=\"\""
-			_, stderr, err := util.RunOVNNbctl("set", "load_balancer", lb, key)
-			if err != nil {
-				klog.Errorf("Error in deleting endpoints for lb %s, "+
-					"stderr: %q (%v)", lb, stderr, err)
-			}
-		} else {
-			_, stderr, err := util.RunOVNNbctl("remove", "load_balancer", lb,
-				"vips", quotedHostPort)
+		// apply reject ACL if necessary before deleting endpoints (avoids unwanted traffic events hitting OVN/OVS)
+		if ovn.svcQualifiesForReject(svc) {
+			aclUUID, err := ovn.createLoadBalancerRejectACL(lb, svc.Spec.ClusterIP, svcPort.Port, svcPort.Protocol)
 			if err != nil {
-				klog.Errorf("Error in deleting lb %s, stderr: %q (%v)", lb, stderr, err)
+				klog.Errorf("Failed to create reject ACL for load balancer: %s, error: %v", lb, err)
 			}
+			klog.V(5).Infof("Reject ACL created for load balancer: %s, %s", lb, aclUUID)
+		}
+
+		// clear endpoints from the LB
+		err := ovn.configureLoadBalancer(lb, svc.Spec.ClusterIP, svcPort.Port, "")
+
+		if err != nil {
+			klog.Errorf("Error in deleting endpoints for lb %s: %v", lb, err)
 		}
+		vip := util.JoinHostPortInt32(svc.Spec.ClusterIP, svcPort.Port)
+		ovn.RemoveServiceEndpoints(lb, vip)
 
```

Comment:
@trozet might as well remove this blank line while you're here.

Reply:
ok